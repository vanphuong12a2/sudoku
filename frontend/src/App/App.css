header {
    text-align: center;
    margin: auto;
    max-width: 600px;
}

.error {
    text-align: center;
}

.error-message {
    display: inline-flex;
    vertical-align: middle;
}