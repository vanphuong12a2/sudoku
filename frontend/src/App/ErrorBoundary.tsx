import * as React from 'react';

interface Props {
    children: React.ReactNode
}

interface State {
    hasError: boolean
}

class ErrorBoundary extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {hasError: false};
    }

    public componentDidCatch(error: Error, info: React.ErrorInfo) {
        this.setState({hasError: true});
    }

    public render(): React.ReactNode {
        if (this.state.hasError) {
            return <div>Something went wrong. Please refresh the page :)</div>;
        }
        return this.props.children
    }
}

export default ErrorBoundary;