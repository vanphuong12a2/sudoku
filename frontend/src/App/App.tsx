import React, {Component} from 'react';
import './App.css'
import GameContainer from '../Game/GameContainer';

class App extends Component {
    render() {
        return (
            <React.Fragment>
                <header>
                    <h3>Sudoku 26</h3>
                </header>
                <GameContainer/>
            </React.Fragment>
        );
    }
}

export default App;
