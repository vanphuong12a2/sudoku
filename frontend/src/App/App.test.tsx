import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {create} from 'react-test-renderer';

jest.mock('../common/boardFunctions', () => {
    return {
        generateBoard: () => {
            return [
                [undefined, 2, 3, 4, 5, 6, 7, 8, 9],
                [1, 2, 3, 4, 5, 6, 7, 8, 9],
                [1, 2, 3, 4, 5, undefined, 7, 8, 9],
                [1, 2, 3, 4, 5, 6, 7, 8, 9],
                [1, 2, 3, 4, 5, 6, 7, 8, 9],
                [1, 2, 3, 4, 5, 6, 7, 8, 9],
                [1, undefined, 3, 4, 5, 6, 7, 8, 9],
                [1, 2, 3, 4, 5, 6, 7, 8, 9],
                [1, 2, 3, 4, 5, 6, 7, 8, 9]
            ];
        }
    };
});


describe('<App />', () => {

    describe('Snapshots', () => {
        it('renders correctly', () => {
            const accordion = create(<App/>);
            expect(accordion.toJSON()).toMatchSnapshot();
        });
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })
});