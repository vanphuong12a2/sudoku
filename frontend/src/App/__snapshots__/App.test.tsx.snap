// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshots renders correctly 1`] = `
Array [
  <header>
    <h3>
      Sudoku 26
    </h3>
  </header>,
  <div
    className="game-container"
  >
    <table
      className="main-board"
    >
      <tbody>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
          <td>
            <input
              className="writable"
              maxLength={1}
              onChange={[Function]}
              readOnly={false}
              value=""
            />
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="control-area"
    >
      <fieldset>
        <legend>
          Game
        </legend>
        <a
          className="waves-effect waves-light btn-small"
          onClick={[Function]}
        >
          Refresh
        </a>
        <a
          className="waves-effect waves-light btn-small"
          onClick={[Function]}
        >
          New game
        </a>
      </fieldset>
      <fieldset>
        <legend>
          Solver
        </legend>
        <a
          className="waves-effect waves-light btn-small"
          onClick={[Function]}
        >
          Clear all
        </a>
        <a
          className="waves-effect waves-light btn-small"
          onClick={[Function]}
        >
          Show solution
        </a>
      </fieldset>
    </div>
  </div>,
]
`;
